### PR TITLE
Add ability to archive/unarchive a tree

### PR DESCRIPTION
**Pull request by kennethlien**
## Summary
Add in an `active` tag to trees to represent archived/unarchived trees. Enable searching only active trees/all trees. Add toggle to archive/unarchive a tree in edit screen.
## Test Plan
Test buttons and make sure they sync up with database records. Automate testing.
## Other
Known issue, the button to switch between archived and unarchived trees can suffer from a desync? It doesn't match what is actually displayed.

## Screenshots and Tests
- if your feature is visual, attach before/after screenshots and ensure that tests still pass
CC: @mylesdomingo

### CODE REVIEW

**Comment by kennethlien** [comment]
imageimageimageimage

File: src/screens/LoginScreen.tsx [hunk 1]
```diff
@@ -94,6 +94,7 @@ export default function Login({ navigation }) {
         </TouchableOpacity>
       ) : undefined}
       {attemptInvisibleVerification && <FirebaseRecaptchaBanner />}
+      <Button title="BYPASS" onPress={() => navigation.navigate('Root')} />
```
Comment (mylesdomingo) [inline src/screens/LoginScreen.tsx:97]: just set attemptInvisibleVerification to true, i think

https://docs.expo.dev/versions/latest/sdk/firebase-recaptcha/

also if you do this, do it as a separate PR please

File: src/screens/SearchScreen.tsx [hunk 2]
```diff
@@ -2,19 +2,21 @@ import React, { useEffect, useState } from 'react';
...
 export default function SearchScreen({
   navigation,
 }: RootTabScreenProps<'Search'>) {
   const [searchQuery, setSearchQuery] = useState('');
   const [trees, setTrees] = useState<Tree[]>([]);
-  const filtered = trees
+  const [active, setActive] = useState<boolean>(true);
+
+  let filtered = trees
     .filter(
```
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:20]: we're already filtering client side here to match search queries, do we need another read operation using a .where clause? we can do the same checking if its active or not

File: src/screens/SearchScreen.tsx [hunk 3]
```diff
@@ -49,6 +61,14 @@ export default function SearchScreen({
           onChangeText={onSearchChange}
           value={searchQuery}
         />
+        <Button
+          onPress={() => {
+            setActive(!active);
```
Comment (mylesdomingo) [inline src/screens/SearchScreen.tsx:66]: you need to run refreshTrees in a useEffect, setActive is an async function and will not complete by the time you refresh

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

File: src/database/firebase.tsx [hunk 4]
```diff
@@ -71,6 +71,21 @@ export const getAllTrees = async (): Promise<Tree[]> => {
   }
 };
 
+/**
+ * getActiveTrees
```
Comment (wangannie) [inline src/database/firebase.tsx:75]: nit: add details for consistency

File: src/screens/TreeDetailsScreen.tsx [hunk 5]
```diff
@@ -83,6 +84,10 @@ export default function TreeDetailsScreen({
         style={styles.input}
         value={entry.id}
       />
+      <Button
```
Comment (wangannie) [inline src/screens/TreeDetailsScreen.tsx:87]: Very confusing that this is a button & clickable when not in edit mode -- i was clicking on it expecting it to save to firebase. Ik this is all temporary but for the sake of bug bash, change to regular text like Status: Active

File: src/screens/SearchScreen.tsx [hunk 6]
```diff
@@ -42,13 +44,21 @@ export default function SearchScreen({
   return (
     <ViewContainer>
       <ScrollView>
+        <Title>Hello</Title>
```
Comment (wangannie) [inline src/screens/SearchScreen.tsx:47]: Is this here by mistake?

**Review by wangannie — APPROVED** [review]
Commented on some quick fixes but good to merge once those are addressed!

**Pull request closed without merging**